lib/charms/layer/apache_zeppelin.py

```diff
@@ -13,15 +24,37 @@ class Zeppelin(object):
 
     :param DistConfig dist_config: The configuration container object needed.
     """
+    @classmethod
+    def get(cls):
+        """
+        Dynamically instantiate this or a subclass, to allow other layers
+        to override portions of this implementation.
+        """
+        impl = layer.options('apache-zeppelin')['implementation_class']
+        module_name, cls_name = impl.rsplit('.', 1)
+        cls = getattr(import_module(module_name), cls_name)
+        return cls()
```

Comment:
I agree, it's not pretty.  The issue is that the reactive handlers in the base layer don't know anything about layers containing subclasses, so they can't instantiate the correct class.  This was my attempt to enable the base layer to be aware of the desired subclass, but I'm more than open to a better way to accomplish this.